src/Application/Command/ReleaseCommand.php

```diff
@@ -84,22 +84,28 @@ public function execute(InputInterface $input, OutputInterface $output): int
             sprintf('No valid release branch found for version %s', $releaseVersion->fullReleaseName())
         );
 
-        ($this->commitChangelog)($repositoryPath, $releaseVersion, $releaseBranch);
-
-        $changelog = ($this->createChangelogText)(
+        $changelogReleaseNotes = ($this->createChangelogText)(
             $milestone,
             $repositoryName,
             $releaseVersion,
             $releaseBranch,
             $repositoryPath
         );
 
+        ($this->commitChangelog)($changelogReleaseNotes, $repositoryPath, $releaseVersion, $releaseBranch);
```

Comment:
I like how this now receives a pre-computed changelog 👍

Reply:
I thought you might. :smile:

Reply:
This made me happy as well. The "commit, then create" from before worked, and was necessary based on what we were doing, but this makes sooooo much more sense.